pkgs/top-level/linux-kernels.nix

```diff
@@ -401,7 +413,7 @@ in {
 
     lkrg = callPackage ../os-specific/linux/lkrg {};
 
-    v4l2loopback = callPackage ../os-specific/linux/v4l2loopback { };
+    v4l2loopback = if !isCachyMain then callPackage ../os-specific/linux/v4l2loopback { } else null;
```

Comment:
Instead of this, mark them as broken.

Reply:
I tried https://github.com/NixOS/nixpkgs/blob/308bc35b256e446dd4869dd3a7eaa82bf37e3be4/pkgs/os-specific/linux/v4l2loopback/default.nix#L42 but Nix still seems to attempt to build this derivation

Reply:
I tried https://github.com/NixOS/nixpkgs/blob/308bc35b256e446dd4869dd3a7eaa82bf37e3be4/pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix#L36 but this module still seems to fail to build

Reply:
That's odd, isCachy is false:

```
nix-repl> :p legacyPackages.x86_64-linux.linuxPackages_cachyos_latest.mbp2018-bridge-drv.meta.broken
false

nix-repl> :p legacyPackages.x86_64-linux.linuxPackages_cachyos_latest.kernel.isCachy
false

nix-repl> :p legacyPackages.x86_64-linux.linuxPackages_cachyos_latest.kernel.isCachyMain
false
```

Reply:
Found it. My bad, I forgot to add isCachy to passthru in my suggested commit that added this bool.

Reply:
```diff
diff --git a/pkgs/os-specific/linux/kernel/generic.nix b/pkgs/os-specific/linux/kernel/generic.nix
index a270e6c944b5..5ea8f2f6c988 100644
--- a/pkgs/os-specific/linux/kernel/generic.nix
+++ b/pkgs/os-specific/linux/kernel/generic.nix
@@ -231,7 +231,7 @@ kernel.overrideAttrs (finalAttrs: previousAttrs: {
 
   passthru = previousAttrs.passthru or { } // basicArgs // {
     features = kernelFeatures;
-    inherit commonStructuredConfig structuredExtraConfig extraMakeFlags isZen isHardened isLibre;
+    inherit commonStructuredConfig structuredExtraConfig extraMakeFlags isCachy isCachyMain isZen isHardened isLibre;
     isXen = lib.warn "The isXen attribute is deprecated. All Nixpkgs kernels that support it now have Xen enabled." true;
 
     # Adds dependencies needed to edit the config:
```

Reply:
Instead of introducing a new variable, can you use `kernel.isCachy && (lib.versionAtLeast kernel.version "6.10")`

Reply:
I also think `isCachyMain` is unnecessary and it might cause some problems in the future.

What matters with modules is the compatibility with kernel versions, not if the kernel is LTS or not. The next LTS kernel will have much more in common with the previous mainline release than the previous LTS release. The checks for the LTS variant might not make much sense at that point, but version checks would stay relevant. 

For example, if some module was incompatible with the previous LTS but was compatible with the previous mainline, why would we assume that it's suddenly broken because the new kernel is a LTS? 

If the previous LTS kernel needed some extra module, but it was irrelevant for the previous mainline because it was simply upstreamed, why would the next LTS suddenly need it? Situations like this would make maintenance a bit annoying because we'd have to go over uses of `isCachyMain` and make sure they're still relevant. Simple `versionAtLeast` checks would make more sense and require less maintenance effort.